tools/tpp-run/MLIRBench.cpp

```diff
@@ -267,14 +294,37 @@ Value MLIRBench::createTimerLoop(unsigned n) {
   builder.setInsertionPointToStart(loop.getBody());
 
   // Call the kernel, ignore output
-  callKernel();
+  auto *call = callKernel();
+  assert(call && "Failed to generate a kernel call");
 
   // Revert insertion point and return the accumulation ID
   builder.setInsertionPointAfter(loop);
   return acc;
 }
 
-Value MLIRBench::getTimerStats(Value acc) {
+Value MLIRBench::getTimerStats(Value accBuf) {
+  // Skip the warmup loops in stats calculation
+  auto dimIdx = builder.create<arith::ConstantIndexOp>(unkLoc, 0);
+  auto len = builder.create<memref::DimOp>(unkLoc, accBuf, dimIdx);
+
+  // The offset is slightly inaccurate as the buffer lenght already includes
```

Comment:
I'm not sure I understand this comment. Isn't the point of the offset to not include the initial runs? And if so, aren't you taking the average of the remaining runs divided by the length - offset?

Reply:
Here it refers to the fact that I apply the same formula to both `n` and `1.1 * n`. Which is technically incorrect as I should solve it always for the original `n` but this simplification works fine here.
To give an example: for `-n 100`, 10 extra warmup iters are added but when calculating how many iters should be skipped for average calculation, it will skip first 11 iters.
This is mostly because I wanted to avoid floating math or any extra info tracing within the benchmark object.

Overall, all the warmup iters are skipped and sometimes a few extra iters.

Reply:
Why can't you just add `11` to `100`, run `111` times, then take the tail `100` measurements and then divide by `100`?

Reply:
That's what I do. It's how I compute this addition and tail that matters:
* program launch `-n 100` - measure over 100 iters
* run this kernel - how many iters we need to execute? 100 + 0.1 * 100 = 110; run 110 iters
* compute my average - how many iters we need to skip? len(deltas) = 110, 0.1 * 110 = 11; skip first 11 deltas

It skips 1 delta too many as I don't want to bother with all the int -> float -> int conversions for better accuracy.

Reply:
Alright, I'll just make these methods a bit less generic, hide the num iters as internal benchmark state, and remove these runtime computations.

Reply:
Adapted and added tests.